pegasus_wrapper/workflow.py

```diff
@@ -67,6 +71,21 @@ class WorkflowBuilder:
     in the workflow graph. We compute this based on a job signature
     and only actually schedule the job once.
     """
+    _added_files: Set[File] = attrib(init=False, factory=set)
+
+    _replica_catalog: Path = attrib(init=False)
+
+    def __attrs_post_init__(self) -> None:
+        # Do I need to check if it exists already? Or will users manually remove
```

Comment:
Prefer to do this work in the `from_parameters` factory method and just pass in the path to the replica catalog to the object constructor.